### DISCUSSION TITLE
Tutorial for Forecasting Problem

### DISCUSSION THREAD

**Discussion by HariWu1995** [timeseriesAI/tsai#125] [Ideas]
I see that in your source code, there are 2 datasets **Sunspots** and **Weather** for **forecasting** problem. Could you create a **demo** notebook for 1 of those? 
I would like to know how to **build a dataset** and which of your **models** are suitable for this problem?
Thank you in advance.

**Comment by vrodriguezf** [comment]
correct me if I'm wrong @oguiza , but any model that ends with `Plus`, like `InceptionTimePlus` is suitable for custom 3d heads, and therefore, suitable for forecasting tasks. 

To build the X,y arrays for the dataset you may want to have a look at the [sliding Window](https://timeseriesai.github.io/tsai/data.preparation.html#SlidingWindow]) functionality

Best!

**Comment by oguiza** [comment]
Hi @HariWu1995,
I've been working on 1-2 forecasting tutorials with `tsai` for some time, but I've been very busy lately. There are also multiple variations you can implement, since it's a complex, broad area. 
I'm planning to release a single-step (univariate & multivariate) forecasting notebook and then a multi-step notebook. Hopefully this will provide you some insights on how to perform this type of task with `tsai`. 
I cannot commit though to any particular timing around this. I just need to find a few hours to finalize it. 
The key to performing a forecast though is to transform the data to a supervised task. And to do that you will likely need to use the SlidingWindow functionality that @vrodriguezf mentioned before.

**Reply by farconada** [reply]
Hope to see that notebooks soon

**Reply by mandar-avhad** [reply]
HI, is the notebook for multistep forecasting been released?

**Comment by HariWu1995** [comment]
@vrodriguezf  @oguiza 
Thank you for your support! I have overcome this problem by my own function. No offense but your function hasn't zero-padding yet, so I need to write my own.

**Comment by oguiza** [comment]
I've moved this into discussions since it's more a request than an actual issue.

**Comment by diegoquintanav** [comment]
Hi! 

Apart from using a rolling window, what are other changes that need to be made? Thanks

I've seen there is a `TSForecasting` transformation that is basically the same as `TSRegression`, and I wonder what the difference is at the implementation level. In other words, if I understand forecasting as an extrapolation task whereas regression as an interpolation task, I don't see in the examples how this is being implemented.
